src/main/java/org/testng/internal/TestResult.java

```diff
@@ -87,7 +87,7 @@ public void init (IClass testClass,
       if (m_instance instanceof ITest) {
         m_name = ((ITest) m_instance).getTestName();
       }
-      else if (testClass.getTestName() != null && !testClass.getTestName().isEmpty()) {

```

Comment:
This was the fix for #922. I removed it because it's no longer necessary with this fix, and because it could actually cause issues if a user intentionally specifies a blank test name (using `ITest#getTestName()`).